### PR TITLE
29 api how to upload meida to noition using notion api

### PR DESCRIPTION
**Pull request by caoluan7238**
Hi, sorry for the late reply. This is my first time working with an API so it took me some time.
So I’ve been trying to figure out base on the tutorials, I was able to make some post the informations (title, url, status,did,wallet address) onto a notion page through the notion API. I wasn’t sure how these information are to be inputted through notion api so I just made them add onto the notion database.
For the env file it needs the notion key and notion database id that can be gathered from notion integration.
Please find any adjustments needed in this pull request
I am also fairly new to Github so please correct me of any mistakes

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+const {Client} = require('@notionhq/client')
+
```
[the remark this answers is not in the document]
Comment (caoluan7238) [reply index.js:2]: Hi, can you check my latest changes? I'm not sure I did the right things

File: index.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  }
+  info.push(data)
+  
+  createPage()
+}
+
+
+async function createPage() {
+  
+  for (let content of info) {
```
[the remark this answers is not in the document]
Comment (caoluan7238) [reply index.js:24]: the media info meaning the titles, file urls, pre approval status, did and wallet address? or did you mean something else

File: api/src/router.js [hunk 3]
```diff
@@ -16,6 +16,7 @@ const { Action, Resource } = require("./util/constant");
 const validates = require("./validates/validates");
 const loginState = require("./middlewares/login_state_manager");
 
+const {Client} = require('@notionhq/client')
```
Comment (hanchenglin) [inline api/src/router.js:19]: @caoluan7238 Hi! 
Here do not need import Client.
Please endpoint for your API

For example:
```
router.post(
    "/upload-media",
    verifyIdToken,
    verifyCSRFToken,
    notion.createNotionPage
);
```

File: api/src/controllers/notion.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const {Client} = require('@notionhq/client')
+
+const notion = new Client({auth: process.env.NOTION_KEY})
+
+const info = []
+
+async function getInfo(t,u,c,w) {
```
Comment (hanchenglin) [inline api/src/controllers/notion.js:7]: @caoluan7238 Here is POST method. Please rename method name

File: api/src/services/notion.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            }
+          }
+      }]
+    })
+    
+  }
+  
+}
+
+getInfo('title',
```
Comment (hanchenglin) [inline api/src/services/notion.js:70]: @caoluan7238 Please add exports too.

module.exports = {
  xxxMethod,
};

File: index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (hanchenglin) [reply index.js:2]: @caoluan7238 Thank you! I think your source file level is good!
Have you investigated what information is needed to create logic and call the Notion API?

File: index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (hanchenglin) [reply index.js:24]: @caoluan7238 Yes, Please organize what properties are needed to match the Notion API.

File: index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (caoluan7238) [reply index.js:2]: @hanchenglin Thank you. If it's possible please give me some more time, It'll take me some time i'm still investigating it

[the remark this answers is not in the document]
Comment (hanchenglin) [reply index.js:2]: @caoluan7238 I understand.

[the remark this answers is not in the document]
Comment (caoluan7238) [reply index.js:2]: @hanchenglin  hi! my apology for a slightly late reply. I did some investigations on how to make the logic and call the Notion API and it seems that we'll need to gather the notion API integration key as well as the database key from notion API https://www.twilio.com/blog/manipulate-notion-database-using-node-js . It also seems that the properties of the page created also has to be manually created on Notion.
Please let me know what I should investigate further, thank you very much

[the remark this answers is not in the document]
Comment (hanchenglin) [reply index.js:2]: @caoluan7238 Thank you! 
You can get a Notion API key and run test code in your localhost.

[the remark this answers is not in the document]
Comment (hanchenglin) [reply index.js:2]: @caoluan7238 Hi! What stutus is?
For example
- Get Notion API ->100%
- Create page with test data ->100%
- ...

[the remark this answers is not in the document]
Comment (caoluan7238) [reply index.js:2]: @hanchenglin hello. So I've ran some test by making my own notion account, obtaining the notion api key then try running the code on a localhost from glitch.com (I didn't know any platform to use so please tell me which i can use to make the http requests). The code worked and I was able to create a page on notion. please check the attached photos for my progress
image
image

[the remark this answers is not in the document]
Comment (hanchenglin) [reply index.js:2]: @caoluan7238 Thank you for you test.
Can you test your API with [Postman](https://www.postman.com/) or [Talend API Tester](https://chrome.google.com/webstore/detail/talend-api-tester-free-ed/aejoelaoggembcahagimdiliamlcdmfm?hl=ja) .

[the remark this answers is not in the document]
Comment (caoluan7238) [reply index.js:2]: @hanchenglin   understood, I will try to test the API with those 2 platforms

[the remark this answers is not in the document]
Comment (caoluan7238) [reply index.js:2]: @hanchenglin hi! my apology for the slight delay, just want to keep update on the progress. I am currently figuring out how to test the talend api tester so give me a few more days till i can finish the test. For the postman, it seems that it is a dedicated platform that requires payment plan ? (please let me know if there is a way i can use it for free)

[the remark this answers is not in the document]
Comment (caoluan7238) [reply index.js:2]: @hanchenglin  hi! After watching some tutorials, I was able to test out the API on Postman.
image
image

[the remark this answers is not in the document]
Comment (hanchenglin) [reply index.js:2]: @caoluan7238 Hi! Thank you for your test.
I think you need test your API(Post http://localhost:8888/v1/upload-media), not Notion API directly!

[the remark this answers is not in the document]
Comment (hanchenglin) [reply index.js:2]: @caoluan7238 Hi! You can run your app in your localhost.

api/.env.development was need config. Ask on slack if you need it.

- run api
```
cd api
yarn
yarn load_data

yarn start
```

- run frontend
```
cd frontend
yarn

yarn serve
```

- access frontend
http://localhost:8080

[the remark this answers is not in the document]
Comment (caoluan7238) [reply index.js:2]: hi @hanchenglin I need some help figuring out where to start. i have changed the port to http://localhost:8888/v1/upload-media and is getting the error Error: connect ECONNREFUSED. What should i put in the body section for Postman. Sorry, please walk me through, I have never tested an api before.

[the remark this answers is not in the document]
Comment (hanchenglin) [reply index.js:2]: @caoluan7238 Hi! I am refatoring this issue.
If you want to test this API, you need start API.

[the remark this answers is not in the document]
Comment (caoluan7238) [reply index.js:2]: @hanchenglin understood, thank you!  I am looking up on the web for some instructions. I will try to test the API by tonight and keep you updated with the progress.

[the remark this answers is not in the document]
Comment (hanchenglin) [reply index.js:2]: @caoluan7238 Hi! Your localhost can run frontend and API?

[the remark this answers is not in the document]
Comment (caoluan7238) [reply index.js:2]: @hanchenglin sorry, would it be possible to give me some times till tomorrow to try to figure things out? I found a tutorial and got my ide set up to test the api on postman now, ill test the api and front end by tomorrow and keep you updated

[the remark this answers is not in the document]
Comment (caoluan7238) [reply index.js:2]: @hanchenglin im really sorry for delaying so many times, im still learning how to do these things so it takes me a lot of time

[the remark this answers is not in the document]
Comment (hanchenglin) [reply index.js:2]: @caoluan7238 Thank you! I will refactor the upload media API weekend. I will share how to run app in your localhost.

[the remark this answers is not in the document]
Comment (hanchenglin) [reply index.js:2]: @caoluan7238 Let me merge this PR first.

[the remark this answers is not in the document]
Comment (caoluan7238) [reply index.js:2]: @hanchenglin understood, should I use the develop branch to test the API?

**Pull request merged**